tests/test_serialised_model.py

```diff
@@ -70,7 +79,12 @@ def foo(self):
     with pytest.raises(AttributeError) as e:
         assert Custom({"foo": "NOPE"}).foo == "bar"
```

Comment:
this test confuses me

`Custom({})` fails because `foo` is required to be passed in because it's in `ALLOWED_PROPERTIES`.

`Custom({"foo": "NOPE"})` fails because there's no `foo.setter` attribute

i'm not sure this is testing any behaviour, other than "you can create a SerialisedModel that is fundamentally broken and will never work" 🤔

can we just delete the whole test? or rework it? Is there any unusual behaviour/interactions we might want to check around properties? I know 

(unrelatedly i only looked at it because on 3.10 the error is `can't set attribute "foo"` so doesn't match either case)

Reply:
Tried to explain and clean up in 1c14d165e6b102dc827e08fd4ff18837822510f9